routes/weatherProviders/OWM.ts

```diff
@@ -87,8 +89,53 @@ async function getOWMWeatherData( coordinates: GeoCoordinates ): Promise< Weathe
 	return weather;
 }
 
+async function getOWMEToData( coordinates: GeoCoordinates ): Promise< EToData > {
+	const OWM_API_KEY = process.env.OWM_API_KEY,
+		forecastUrl = "http://api.openweathermap.org/data/2.5/forecast?appid=" + OWM_API_KEY + "&units=imperial&lat=" + coordinates[ 0 ] + "&lon=" + coordinates[ 1 ];
+
+	// Perform the HTTP request to retrieve the weather data
+	let forecast;
+	try {
+		forecast = await httpJSONRequest( forecastUrl );
+	} catch (err) {
+		// Indicate ETo data could not be retrieved if an API error occurs.
+		return undefined;
+	}
+
+	// Indicate ETo data could not be retrieved if the forecast data is incomplete.
+	if ( !forecast || !forecast.list || forecast.list.length < 8 ) {
+		return undefined;
+	}
+
+	// Take a sample over 24 hours.
+	const samples = forecast.list.slice( 0, 8 );
+
+	const cloudCoverInfo: CloudCoverInfo[] = samples.map( ( window ): CloudCoverInfo => {
+		return {
+			startTime: moment.unix( window.dt ),
+			endTime: moment.unix( window.dt ).add( 3, "hours" ),
+			cloudCover: window.clouds.all / 100
+		};
+	} );
+
+	return {
+		weatherProvider: "OWM",
+		minTemp: samples.reduce( ( min, window ) => Math.min( min, window.main.temp ), 1000),
+		maxTemp: samples.reduce( ( max, window ) => Math.max( max, window.main.temp ), -1000),
+		minHumidity: samples.reduce( ( min, window ) => Math.min( min, window.main.humidity ), 100),
+		maxHumidity: samples.reduce( ( max, window ) => Math.max( max, window.main.humidity ), 0),
+		solarRadiation: approximateSolarRadiation( cloudCoverInfo, coordinates ),
+		// Assume wind speed measurements are taken at 2 meters.
+		windSpeed: samples.reduce( ( sum, window ) => sum + window.wind.speed, 0) / samples.length,
+		dayOfYear: moment().dayOfYear(),
```

Comment:
@Derpthemeus can I just check this. OWM is a rolling 24-hour forecast whereas `dayOfYear` is a specific day. Are we mixing two worlds and is the difference material in `calculateETO()` ? Do we need to either embrace the rolling 24-hours or revert to calendar days ?

Reply:
We are currently using a fixed calendar window in Dark Sky and a rolling 24 hour window in OWM. Although this inconsistency isn't ideal, `calculateETo()` should still be accurate for whatever 24 hour window it's given. I think we will want to eventually want to favor the fixed calendar day window approach for ETo once we implement a full water balancing system since it will be easier to work with (by using full calendar days, we won't need to worry about retrieving data at the same time each day).

Reply:
Yep, I understand the DarkSky and OWM difference (PWS is calendar as well). So your suggesting that we should revert back to calendar calculations rather than rolling 24-hours at some point.

My observation was merely that an OWM forecast based call at 23:59 would use declination and sun_distance calc from today rather than tomorrow. Now I have no idea if that is material ! :) !